README.md

```diff
@@ -1,2 +1,91 @@
+
+* `channel`: *Required.* The channel name to be used by the bot.
+
+* `bot_name`: *Optional.*, *default `bender`* The bot name will be used to identify and filter messages. All messages must be addressed to the bot `@bot_name some message`.
+
+* `grammar`: *Optional.* if not defined bender will respond to all mentions `@bot_name`. If grammar is defined bender will respond to messages matching the regex expression. Use [python regular expression](https://docs.python.org/2/library/re.html) syntax. See [examples](#Examples) for inspiration.
+
+* `template`: *Optional.* A string that will be evaluated and written to `template_filename` can be used as an input file for further jobs in the pipeline.
+
+* `template_filename`: *Optional.* *default `template_file.txt`* the file name for a generated template.
+
+## Behavior
+
+### `check`: Check for new messages that match the rules
+
+Check will report a new version `timestamp` if messages
+
+1: Are addressed to `@bot_name`.
+
+2: If `grammar` is defined it matches the regex of the message.
+
+#### `check Parameters`
+
+Check accepts no params.
+
+### `in`: Get the message
+
+A file **bender.json** will be created with the message payload.
+
+if `template` is defined it will be evaluated and written `template_filename`. For more info on syntax check [Template section](#Template) for example usage check the [examples](#Examples)
+
+#### `in Parameters`
+
+In accepts no params.
+
+### `out`: Reply to original message thread
+
+Sends a message as a thread to the original message sent by the user.
+
+#### `out Parameters`
+
+* `path`: *Required.* The path of the resource name. To fetch original message ts
+
+* `reply`: *Required.* The content of the reply. This can be also a template.
+
+### Template
+
+Under the hood the template use python [Jinj2](http://jinja.pocoo.org/docs/2.9/).
+
+* `variables`: You can access variables using curly braces `{{ VARIABLE_NAME }}`.
+   All environmental variables are accessible through `{{ ENV['PATH'] }}`. Concourse exposes some metadata info like job_name, build and ... for more info [check concourse website](https://concourse.ci/implementing-resources.html#resource-metadata)
+   The regex groups are accessible to the template engine as `{{ regex }}` if you used *subgroups* in your expression you can access each group with index `{{ regex[0] }}`. If you used *named subgroups* you can access them as dictionary `{{ regex['name'] }}`.
+
+* `white spaces`: You can use `\n` for new lines `\t`.
+
+* `filters`: You can use Jinja2 [built filters](http://jinja.pocoo.org/docs/2.9/templates/#builtin-filters).
+
+## Tips
+
+* Test your regex using something like [regexr.com](http://regexr.com/), It is better to use [named subgroups](http://www.regular-expressions.info/brackets.html).
+
+* Note that you need to escape `\` with another `\\` in your expression.
```

Comment:
Maybe rephrase as double-escape.

Reply:
Because now it sounds like you need to escape `\` with `\\` which will make `\\\` \\\\\ \\\\\\\\ \\\ \\\\